src/web/lib/storage.js

```diff
@@ -17,16 +17,33 @@ const generateThemeKey = () =>
   `${Date.now()}-${Math.floor(Math.random() * 1000)}`;
 
 function putTheme(key, theme) {
-  log("putTheme", key, theme);
-  const storageKey = themeStorageKey(key);
-  localStorage.setItem(
-    storageKey,
-    JSON.stringify({
-      theme,
-      modified: Date.now()
-    })
-  );
-  notifySelfForStorage(storageKey);
+  checkDuplicateTheme(theme, (isThemeDuplicate) => {
+    if (!isThemeDuplicate) {
+      log("putTheme", key, theme);
+      const storageKey = themeStorageKey(key);
+      localStorage.setItem(
+        storageKey,
+        JSON.stringify({
+          theme,
+          modified: Date.now()
+        })
+      );
+      notifySelfForStorage(storageKey);
+    }
+  });
+}
+
+function checkDuplicateTheme(theme, callback) {
+  const themesList = listThemes();
+  let isThemeDuplicate = false;
+  for (const key of Object.keys(themesList)) {
+    isThemeDuplicate = themesEqual(theme, themesList[key].theme);
+    if (isThemeDuplicate) {
+      callback(isThemeDuplicate);
```

Comment:
From the looks of this, wouldn't the callback be called twice here? Once on L42 if a match was found, and again on L46 after the `break` statement breaks out of the `for..of` loop.